File: stream/src/main/java/bsky4j/stream/api/atproto/SyncResource.java
package kbsky.stream.api.atproto;

import kbsky.stream.api.entity.atproto.sync.SyncSubscribeReposRequest;
import kbsky.stream.util.StreamClient;


/**
 * ATProtocol/Sync
 * <a href="https://atproto.com/lexicons/com-atproto-sync#comatprotosyncsubscriberepos">Reference</a>
 */
public interface SyncResource {

    /**
     * [WIP]
     * Subscribe to repo updates
     */
    StreamClient subscribeRepos(SyncSubscribeReposRequest request);
}

File: stream/src/commonMain/kotlin/work/socialhub/kbsky/stream/api/atproto/SyncResource.kt
package work.socialhub.kbsky.stream.api.atproto

import work.socialhub.kbsky.stream.api.entity.atproto.sync.SyncSubscribeReposRequest
import work.socialhub.kbsky.stream.util.StreamClient

/**
 * ATProtocol/Sync
 * [Reference](https://atproto.com/lexicons/com-atproto-sync#comatprotosyncsubscriberepos)
 */
interface SyncResource {
    /**
     * [WIP]
     * Subscribe to repo updates
     */
    fun subscribeRepos(
        request: SyncSubscribeReposRequest
    ): StreamClient
}
